main/pipeline/pipeline.cc

```diff
@@ -274,6 +275,18 @@ incrementalResolve(core::GlobalState &gs, vector<ast::ParsedFile> what,
             // Cancellation cannot occur during incremental namer.
             ENFORCE(!canceled);
 
+            if (opts.runLSP && gs.lspQuery.kind == core::lsp::Query::Kind::NONE) {
```

Comment:
I pass it as a param from a call site, but I'll need that info one way or another. The reason is we still want to flush errors only once for CLI mode. Otherwise we would see duplicated errors in the cli output

Reply:
So we already have two implementations of the abstract `core::ErrorFlusher` class:

- `ErrorFlusherStdout` for CLI mode
- `ErrorFlusherLSP` for LSP

Instead of depending on `opts.runLSP` in more places, it would be nice to use the existing abstractions to encapsulate any LSP-specific logic.

Reply:
It's not exactly clear to me how to do it in this case. 
Moving error cache management into `core::ErrorFlusher ` would hit the same issues as moving it into `core::ErrorQueue`. 

I guess I could try casting current `core::ErrorFlusher` to `ErrorFlusherLSP ` instead of checking for `opts.runLSP`, but it feels very similar to me